### PR TITLE
Putmethod

### PR DESCRIPTION
**Pull request by jeffrey20122021**

### CODE REVIEW

**Review by holgiosalos — COMMENTED** [review]
You need to update this branch with the changes of the previous PR.

1. Squash and merge PR #6 
2. In your local, checkout to main/master branch
3. git pull
4. checkout to this (putmethod) branch
5. merge the changes in master/main branch 
6. push those changes to this (putmethod) branch

**Comment by jeffrey20122021** [comment]
> You need to update this branch with the changes of the previous PR.
> 
> 1. Squash and merge PR #6
> 2. In your local, checkout to main/master branch
> 3. git pull
> 4. checkout to this (putmethod) branch
> 5. merge the changes in master/main branch
> 6. push those changes to this (putmethod) branch

Done

**Comment by jeffrey20122021** [comment]
Done

**Review by holgiosalos — COMMENTED** [review]
changes in the following files should not be part of this PR, since those changes were already in #6, please merge main/master branch with this branch:

1. test/GithubApi.Repositories.test.js
2. package.json
3. .eslintrc.yml
4. README.md

**Comment by jeffrey20122021** [comment]
> changes in the following files should not be part of this PR, since those changes were already in #6, please merge main/master branch with this branch:
> 
> 1. test/GithubApi.Repositories.test.js
> 2. package.json
> 3. .eslintrc.yml
> 4. README.md

done

**Comment by jeffrey20122021** [comment]
done

File: test/GithubApi.Put.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+describe('idempotence', () => {
+  it('PUT ', async () => {
+    const response = await object.put('https://api.github.com/user/following/aperdomob');
+    expect(response.status).to.equal(204);
+    expect(response.data).to.equal('');
+  });
+  it('Following test', async () => {
+    const response = await object.get('https://api.github.com/user/following');
+    expect(response.data.find((user) => user.login === 'aperdomob')).to.not.eql(undefined);
+  });
```
Comment (holgiosalos) [inline test/GithubApi.Put.test.js:18]: with the existing tests, you are covering points 2 and 3, for the step 4, you need to add another `it` to call again the put and then verify that you continue following the user `aperdomob`

Comment (jeffrey20122021) [reply test/GithubApi.Put.test.js:18]: done

File: test/GithubApi.Put.test.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  it('refollowing test', async () => {
+    const response = await object.get('https://api.github.com/user/following');
```
Comment (holgiosalos) [inline test/GithubApi.Put.test.js:27]: please also call the put, the idea to test the idempotence is that, if we perform the same action several times (in this case the call to the put method), the result always will be the same (in this case that we continue following the desired user. So, the steps needed in this `it` section are:

1. Call again the put method (missing)
2. Call again the get method (done)
3. Verify you continue following the user (done)

Comment (jeffrey20122021) [reply test/GithubApi.Put.test.js:27]: done

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**